crates/wasmtime/src/config.rs

```diff
@@ -2423,6 +2485,27 @@ impl Config {
             bail!("cannot disable the simd proposal but enable the relaxed simd proposal");
         }
 
+        if features.contains(WasmFeatures::STACK_SWITCHING) {
```

Comment:
One other piece of validation (which might belong in `Engine::_check_compatible_with_native_host` perhaps) is to ensure that if stack-switching is enabled that `#[cfg(feature = "stack-switching")]` is also enabled. Otherwise that represents a host that has enabled the wasm feature but not compile-time support so it'd probably be good to return a first-class error in such a situation.